### ISSUE TITLE
Validate happens even StructField is nil 

### ISSUE THREAD

**Issue by anh** [typerandom/validator#32]
After reading https://github.com/typerandom/validator/wiki/Tagging#structure I did following example and the result confused me since `nil` structfield was still validated or I did something wrong?

```
{Title:abc ProducedBy:0xc20800a310}
[ProducedBy.Age cannot be less than 18.]

{Title:abc ProducedBy:<nil>}
[ProducedBy.Age cannot be less than 18.]
```

```
package main

import (
    "encoding/json"
    "fmt"
    "github.com/typerandom/validator"
)

type Album struct {
    Title      string    `json:"title" validate:"not_empty"`
    ProducedBy *Producer `json:"produced_by,omitempty"`
}

type Producer struct {
    Age int `json:"age" validate:"min(18)"`
}

func main() {
    jsonAlbum := []byte(`{"title": "abc", "produced_by": { "age": 17}}`)

    var a Album
    if err := json.Unmarshal(jsonAlbum, &a); err != nil {
        fmt.Println(err)
        return
    }

    fmt.Printf("%+v\n", a)
    if err := validator.Validate(a); err.Any() {
        fmt.Println(err)
    }

    jsonAlbum2 := []byte(`{"title": "abc"}`)

    var a2 Album
    if err := json.Unmarshal(jsonAlbum2, &a2); err != nil {
        fmt.Println(err)
        return
    }

    fmt.Printf("\n%+v\n", a2)
    // no error expected
    if err := validator.Validate(a2); err.Any() {
        fmt.Println(err)
    }
}
```

**Comment by rorheden** [comment]
Thanks for reporting this, this is definitely an issue in the library!

I'm going to look into this and get back to you when I have a fix.

Cheers! :beers:

**Comment by rorheden** [comment]
Sorry for the slow reply. This is now fixed in 15e4d81a0a8ad87e99c0ae99e612e675ce3f56a0. Let me know how it works for you.

Closing issue meanwhile.

**Comment by anh** [comment]
It works, tagging with `"nil|not_empty"` now also works.
Thank you for the fix and great validate lib.
